mpsort-mpi.c

```diff
@@ -224,31 +259,17 @@ _create_segment_group(struct SegmentGroupDescr * descr, size_t * sizes, size_t *
 
     MPI_Comm_rank(descr->Group, &rank);
 
-    struct { 
-        size_t val;
-        int   rank;
-    } leader_st;
-
-    leader_st.val = sizes[ThisTask];
-    leader_st.rank = rank;
+    /* rank with most data in a group is the leader of the group. */
+    descr->group_leader_rank = MPIU_GetLoc(&sizes[ThisTask], MPI_LONG, MPI_MAX, descr->Group);
 
-    MPI_Allreduce(MPI_IN_PLACE, &leader_st, 1, MPI_LONG_INT, MPI_MAXLOC, descr->Group);
```

Comment:
Hang on, isn't the code before just wrong? sizeof(leader_st) == 12 != sizeof(MPI_LONG_INT) == 8?